src/library/scala/collection/Factory.scala

```diff
@@ -91,7 +91,7 @@ trait IterableFactory[+CC[_]] extends Serializable {
     * @param elems  the elements of the created $coll
     * @return a new $coll with elements `elems`
     */
-  def apply[A](elems: A*): CC[A] = from(new View.Elems(elems: _*))
```

Comment:
@szeiger WDYT of this change? Is the `new View.Elems` thing a hangover of an older state of the design, or does it serve some purpose?

One functional change after my patch is that:

```
val ls = List(1, 2, 3)
List.apply(ls: _) eq ls
```

holds (thanks to the optimization in `List.from`), as it used to on 2.12.

Reply:
@retronym Also, does Coll() eq Coll.empty? (for immutable collections)

Reply:
It does after this PR, as the empty varargs argument (`Nil`) has `knownSize == 0`, which enables, for instance:

```scala
object ChampHashMap extends MapFactory[ChampHashMap] {
 ...
  private final val EmptyMap = new ChampHashMap(MapNode.empty, 0, 0)

  def empty[K, V]: ChampHashMap[K, V] =
    EmptyMap.asInstanceOf[ChampHashMap[K, V]]

  def from[K, V](source: collection.IterableOnce[(K, V)]): ChampHashMap[K, V] =
    source match {
      case hs: ChampHashMap[K, V] => hs
      case _ if source.knownSize == 0 => empty[K, V]
      case _ => (newBuilder[K, V] ++= source).result()
    }
```

Reply:
I'll add some unit tests for this.

Reply:
@retronym :+1:

Reply:
Yes, the `View.Elems` wrapper appears unnecessary. Maybe we used it to bridge varargs (old Scala collections) to strawman collections. Some of the other Factory classes also use it. They should be changed, too.

Reply:
Ah, I see you already did that.